ecl/hql/hqlopt.cpp

```diff
@@ -1644,6 +1644,9 @@ void CTreeOptimizer::analyseExpr(IHqlExpression * expr)
         //only look at the filename - not the parent files.
         analyseExpr(expr->queryChild(0));
         return;
+    case no_attr_expr:
```

Comment:
This meant that usage counts were not updated for any attributes on expressions - which caused problems for the loopfirst attribute - causing work to be duplicated.